utils.py

```diff
@@ -352,40 +349,58 @@ def print_attributes(vds, path=None):
+        Get numbered annotations from a VDS variant schema based on their `Number` va attributes.
+    The numbered annotations are returned as a dict with the Number as the key and a list of tuples (field_path, field) as values.
+    All annotations that do not have a Number attribute are returned under the key `None`
+    :param TStruct schema: Input variant schema
+    :param str root: Root path to get annotations (defaults to va)
+    :param bool recursive: Whether to go recursively to look for Numbered annotations in TStruct fields
+    :return: Dictionary containing annotations
+    :rtype: dict of tuple(str, Field)
     """
-    a_annotations = []
-    g_annotations = []
-    dot_annotations = []
-    annotations = []
+    annotations = group_annotations_by_attribute(schema, 'Number', root, recursive)
+    logger.info("Found the following fields:")
+    for k, v in annotations.iteritems():
+        if k is not None:
+            logger.info("{}-based annotations: {}".format(k, ",".join([fields[0] for fields in v])))
+        else:
+            logger.info("Annotations with no number: {}".format(",".join([fields[0] for fields in v])))
+
+    return annotations
+
+def group_annotations_by_attribute(schema, grouping_key , root, recursive = False):
+    """
+    Groups annotations in a dictionnary by the given attribute key.
+    All annotations that do not have a Number attribute are returned under the key `None`
+
+    :param TStruct schema: Input schema
+    :param str root: Root path to get annotations
+    :param bool recursive: Whether to go recursively to look for Numbered annotations in TStruct fields
+    :return: Dictionary containing annotations
+    :rtype: dict of tuple(str, Field)
+    """
+    annotations = defaultdict(list)
+
+    if '.' in root:
+        fields = get_ann_type(root, schema)
+    else:
+        fields = schema
 
-    release_info = get_ann_type(root, vds.variant_schema)
-    for field in release_info.fields:
+    for field in fields.fields:
+        path = '{}.{}'.format(root, field.name)
         if isinstance(field.typ, TArray):
-            if 'Number' in field.attributes:
-                number = field.attributes['Number']
-                if number == "A":
-                    a_annotations.append(field)
-                elif number == "G":
-                    g_annotations.append(field)
-                else:
-                    dot_annotations.append(field)
+            if grouping_key in field.attributes:
+                annotations[field.attributes[grouping_key]].append((path, field))
+        elif recursive and isinstance(field.typ, TStruct):
+            f_annotations = group_annotations_by_attribute(schema, grouping_key, path, recursive)
+            for k,v in f_annotations.iteritems():
+                annotations[k].extend(v)
         else:
-            annotations.append(field)
+            annotations[None].append((path, field))
```

Comment:
Is this gonna work in this case? Previously, we had a `dot_annotations` and an `annotations` to catch things that were arrays of length `.` vs non-arrays.

Reply:
Ohhhhh, I see. So `.` will still be accessed like `annotations['.']` but then `annotations[None]` is the array versions. I think that's fine, but could be documented a bit more.

Reply:
OK! Yeah, I was looking for a key that cannot possibly a String